app/api/threads/list_threads.feature

```diff
@@ -0,0 +1,172 @@
+Feature: Get threads
+  Background:
+    Given there are the following users:
+      | login               |
+      | PresidentConsortium |
+      | PresidentUniversity |
+      | RichardFeynman      |
+      | Mary                |
+      | EtienneKlein        |
+      | Charlotte           |
+      | Baptiste            |
+      | Thibaut             |
+      | DavidBowie          |
+    And there are the following groups:
+      | parent               | name                 |
+      |                      | UniversityConsortium |
+      | UniversityConsortium | University           |
+      | University           | PresidentUniversity  |
+      | University           | FirstYear            |
+      | FirstYear            | Classroom            |
+      | UniversityConsortium | Université           |
+      | Université           | PremièreAnnée        |
+      | PremièreAnnée        | Classe               |
+      |                      | Superstar            |
+    And there are the following group members:
+      | group                | member              |
+      | UniversityConsortium | PresidentConsortium |
+      | University           | RichardFeynman      |
+      | Classroom            | Mary                |
+      | Université           | EtienneKlein        |
+      | PremièreAnnée        | Charlotte           |
+      | PremièreAnnée        | Baptiste            |
+      | PremièreAnnée        | Thibaut             |
+      | Superstar            | DavidBowie          |
+    And there are the following items with permissions:
+      | group        | item                                  | has_validated | can_view                 | can_watch |
+      | Baptiste     | BaptisteCanViewInfo                   |               | info                     |           |
+      | Baptiste     | BaptisteCanViewContent1               |               | content                  |           |
+      | Baptiste     | BaptisteCanViewContent2               |               | content                  |           |
+      | Baptiste     | BaptisteCanViewContentWithDescendants |               | content_with_descendants |           |
+      | EtienneKlein | EtienneKleinHasValidated1             | 1             |                          |           |
+      | EtienneKlein | EtienneKleinHasValidated2             | 1             |                          |           |
+      | EtienneKlein | EtienneKleinHasValidated3             | 1             |                          |           |
+      | EtienneKlein | EtienneKleinHasValidated4             | 1             |                          |           |
+      | EtienneKlein | EtienneKleinHasValidated5             | 1             |                          |           |
+      | EtienneKlein | EtienneKleinHasValidated6             | 1             |                          |           |
```

Comment:
Since "has_validated" is something between an item and a user, it seemed a good idea to put it there. But there is a semantic problem: it doesn't make sense to say that "a group" validated an item, does it?

Maybe we can separate the definitions and add something like:
```
    And there are the following items which have been validated:
      | item                      | member       |
      | EtienneKleinHasValidated1 | EtienneKlein |
      | EtienneKleinHasValidated2 | EtienneKlein |
      | EtienneKleinHasValidated3 | EtienneKlein |
      | EtienneKleinHasValidated4 | EtienneKlein |
      | EtienneKleinHasValidated5 | EtienneKlein |
      | EtienneKleinHasValidated6 | EtienneKlein |
```

Would that be better? Also, is "member" the right name here?

Reply:
> Also, is member the right name here?

"participant" (=team or user) would be the right name (see `participant_id` in the `results` table ... actually to avoid confusion, naming should be close to those use in the db)`

> Would that be better?

It is just related to what you put in your `results` table... couldn't it just be the result table? or a "factory" around this table

Reply:
fixed, added validated in "And there are the following results:"